pkg/processor/timeout/timeout.go

```diff
@@ -98,18 +97,21 @@ func (w EventTimeoutWatcher) watch() {
 							"elapsed", elapsedTime,
 						}
 
-						// if the worker can be restarted, restart it. otherwise time it out
+						if err := triggerInstance.TimeoutWorker(workerInstance); err != nil {
+							w.logger.WarnWithCtx(workerErrGroupCtx,
+								"Error timing out a worker",
+								with...)
+						}
+
+						// if the worker can be restarted, restart it. otherwise shut it completely
 						if workerInstance.SupportsRestart() {
 							w.logger.InfoWithCtx(workerErrGroupCtx, "Restarting worker due to timeout", with...)
 							if err := workerInstance.Restart(); err != nil {
 								with = append(with, "error", err)
 								w.logger.ErrorWithCtx(workerErrGroupCtx, "Can't restart worker", with...)
 							}
 						} else {
-							if err := triggerInstance.TimeoutWorker(workerInstance); err != nil {
```

Comment:
I wonder why was it only called in case restart was not supported in the first place 🤔

Reply:
I guess a bug